packages/contracts/test/exchange/transactions.ts

```diff
@@ -0,0 +1,195 @@
+                    .times(makerAssetFillAmount)
+                    .dividedToIntegerBy(signedOrder.makerAssetAmount);
+                const takerFeePaid = signedOrder.takerFee
+                    .times(makerAssetFillAmount)
+                    .dividedToIntegerBy(signedOrder.makerAssetAmount);
+                expect(newBalances[makerAddress][defaultMakerTokenAddress]).to.be.bignumber.equal(
+                    erc20Balances[makerAddress][defaultMakerTokenAddress].minus(makerAssetFillAmount),
+                );
+                expect(newBalances[makerAddress][defaultTakerTokenAddress]).to.be.bignumber.equal(
+                    erc20Balances[makerAddress][defaultTakerTokenAddress].add(takerAssetFillAmount),
+                );
+                expect(newBalances[makerAddress][zrxToken.address]).to.be.bignumber.equal(
+                    erc20Balances[makerAddress][zrxToken.address].minus(makerFeePaid),
+                );
+                expect(newBalances[takerAddress][defaultTakerTokenAddress]).to.be.bignumber.equal(
+                    erc20Balances[takerAddress][defaultTakerTokenAddress].minus(takerAssetFillAmount),
+                );
+                expect(newBalances[takerAddress][defaultMakerTokenAddress]).to.be.bignumber.equal(
+                    erc20Balances[takerAddress][defaultMakerTokenAddress].add(makerAssetFillAmount),
+                );
+                expect(newBalances[takerAddress][zrxToken.address]).to.be.bignumber.equal(
+                    erc20Balances[takerAddress][zrxToken.address].minus(takerFeePaid),
+                );
+                expect(newBalances[feeRecipientAddress][zrxToken.address]).to.be.bignumber.equal(
+                    erc20Balances[feeRecipientAddress][zrxToken.address].add(makerFeePaid.add(takerFeePaid)),
+                );
+            });
+        });
+
+        describe('cancelOrder', () => {
+            it('should throw if not called by specified sender', async () => {
+                const data = exchange.cancelOrder.getABIEncodedTransactionData(order);
+                const signedTx = makerTransactionFactory.newSignedTransaction(data);
+                return expect(exchangeWrapper.executeTransactionAsync(signedTx, makerAddress)).to.be.rejectedWith(
+                    constants.REVERT,
+                );
+            });
+
+            it('should cancel the order when signed by maker and called by sender', async () => {
+                const data = exchange.cancelOrder.getABIEncodedTransactionData(order);
+                const signedTx = makerTransactionFactory.newSignedTransaction(data);
+                await exchangeWrapper.executeTransactionAsync(signedTx, senderAddress);
+                const res = await exchangeWrapper.fillOrderAsync(signedOrder, takerAddress);
+                const newBalances = await erc20Wrapper.getBalancesAsync();
+                expect(newBalances).to.deep.equal(erc20Balances);
+            });
+        });
+    });
```

Comment:
It could also be worthwhile testing that the `currentSigner` global variable is reset after each transaction.